lib/facter/self_service.rb

```diff
@@ -188,4 +188,19 @@
     # Is puppet_metrics_collector::system configured
     { S0040: PuppetSelfService.service_running_enabled('puppet_system_processes-metrics.timer') }
   end
+
+  chunk(:S0027) do
+    next unless PuppetSelfService.primary?
+    # Check if thundering herd is occuring.
+    thundering_query_output = PuppetSelfService.psql_thundering_herd.split(' ').drop(2).map(&:to_i)
+    thundering_query_output.delete(0)
+    thundering_occured = false
+    minvalue = thundering_query_output.min.to_i
+    maxvalue = thundering_query_output.max.to_i
+    differences = ((maxvalue - minvalue) * 100) / minvalue
+    if differences >= 75
+      thundering_occured = true
+    end
+    { S0027: thundering_occured }
```

Comment:
you can remove the if and the thundering_occured variable and change the line to 

```suggestion
    { S0027: differences >= 75 }
```
and it would have the same result with less code

You also need to flip the logic of the output here, you need this to return True when there is no herd, and false when there is a herd (true indicating  everything is OK)

There are a few ways to do this, like changing it to <= or putting a ! (NOT) in front of the expression